Nodejs/Product/TestAdapterImpl/VisualStudioApp.cs

```diff
@@ -33,41 +31,47 @@ public static VisualStudioApp FromProcessId(int processId)
             return inst;
         }
 
-        public static VisualStudioApp FromEnvironmentVariable(string variable)
+        public static void LogTelemetry(int testCount, Version nodeVersion, bool isDebugging, string testFramework)
         {
-            var pid = Environment.GetEnvironmentVariable(variable);
-            if (pid == null)
-            {
-                return null;
-            }
+            var userTask = new UserTaskEvent("VS/NodejsTools/UnitTestsExecuted", TelemetryResult.Success);
+            userTask.Properties["VS.NodejsTools.TestCount"] = testCount;
+            // This is safe, since changes to the ToString method are very unlikely, as the current output is widely documented.
+            userTask.Properties["VS.NodejsTools.NodeVersion"] = nodeVersion.ToString() ?? "0.0";
+            userTask.Properties["VS.NodejsTools.IsDebugging"] = isDebugging;
+            userTask.Properties["VS.NodejsTools.TestFramework"] = testFramework;
+
+            TelemetryService.DefaultSession?.PostEvent(userTask);
+        }
 
-            if (!int.TryParse(pid, out var processId))
-            {
-                return null;
-            }
+        public static bool AttachToProcessNode2DebugAdapter(int vsProcessId, int port)
+        {
+            var app = FromProcessId(vsProcessId);
+            return app.AttachToProcessNode2DebugAdapter(port);
+        }
 
-            return FromProcessId(processId);
+        public static bool AttachToProcess(Process testProcess, int vsProcessId, Guid portSupplier, string transportQualifierUri)
+        {
+            var app = FromProcessId(vsProcessId);
+            return app.AttachToProcess(testProcess, portSupplier, transportQualifierUri);
         }
 
-        private VisualStudioApp(int processId)
+        public static void DetachDebugger(int vsProcessId)
         {
-            this.processId = processId;
+            var app = FromProcessId(vsProcessId);
+            app.GetDTE().Debugger.DetachAll();
         }
 
-        public void Dispose()
+        private VisualStudioApp(int processId)
         {
-            lock (KnownInstances)
-            {
-                KnownInstances.Remove(this.processId);
```

Comment:
>Remove [](start = 31, length = 6)

Things no  longer get removed from the map?

Reply:
got rid of dispose, since that was never called.